libcfnet/tls_generic.c

```diff
@@ -88,12 +88,6 @@ static int CompareCertToRSA(X509 *cert, RSA *rsa_key)
             TLSErrorString(ERR_get_error()));
         goto ret1;
     }
-    if (EVP_PKEY_type(cert_pkey->type) != EVP_PKEY_RSA)
```

Comment:
According to [docs](https://www.openssl.org/docs/man1.1.0/crypto/EVP_PKEY_type.html) we can switch to `EVP_PKEY_base_id(pkey)`

> Previous versions of this document suggested using EVP_PKEY_type(pkey->type) to determine the type of a key. Since EVP_PKEY is now opaque this is no longer possible: the equivalent is EVP_PKEY_base_id(pkey).